pkg/oc/cli/process/process.go

```diff
@@ -129,84 +165,119 @@ func RunProcess(f kcmdutil.Factory, in io.Reader, out, errout io.Writer, cmd *co
+	if err != nil && !o.local {
+		return err
+	}
+
+	o.outputFormat = kcmdutil.GetFlagString(cmd, "output")
+
+	if o.outputFormat != "describe" {
+		o.Printer, err = o.PrintFlags.ToPrinter()
+		if err != nil {
+			return err
 		}
-		return nil
-	})
-	if len(duplicatedKeys) != 0 {
-		return kcmdutil.UsageErrorf(cmd, fmt.Sprintf("The following parameters were provided more than once: %s", strings.Join(duplicatedKeys.List(), ", ")))
 	}
 
-	filename := kcmdutil.GetFlagString(cmd, "filename")
-	if len(templateName) == 0 && len(filename) == 0 {
-		return kcmdutil.UsageErrorf(cmd, "Must pass a filename or name of stored template")
+	o.usageErrorFn = func(format string, args ...interface{}) error {
+		return kcmdutil.UsageErrorf(cmd, format, args)
 	}
+	return nil
+}
 
-	if kcmdutil.GetFlagBool(cmd, "parameters") {
+func (o *ProcessOptions) Validate(cmd *cobra.Command) error {
+	if o.parameters {
 		for _, flag := range []string{"param", "labels", "output", "output-version", "raw", "template"} {
 			if f := cmd.Flags().Lookup(flag); f != nil && f.Changed {
 				return kcmdutil.UsageErrorf(cmd, "The --parameters flag does not process the template, can't be used with --%v", flag)
 			}
 		}
 	}
 
-	// the namespace
-	namespace, explicit, err := f.ToRawKubeConfigLoader().Namespace()
-	// we only need to fail on namespace acquisition if we're actually taking action.  Otherwise the namespace can be enforced later
-	if err != nil && !local {
-		return err
+	if len(o.templateName) == 0 && len(o.filename) == 0 {
+		return kcmdutil.UsageErrorf(cmd, "Must pass a filename or name of stored template")
+	}
+
+	if len(o.templateName) > 0 && o.local {
+		return kcmdutil.UsageErrorf(cmd, "You may only specify a local template file via -f when running this command with --local")
+	}
+
+	return nil
+}
+
+// RunProcess contains all the necessary functionality for the OpenShift cli process command
+func (o *ProcessOptions) RunProcess(g kcmdutil.Factory) error {
+	duplicatedKeys := sets.NewString()
+	params, paramErr := app.ParseAndCombineEnvironment(o.templateParams, o.paramFile, o.In, func(key, file string) error {
+		if file == "" {
+			duplicatedKeys.Insert(key)
+		} else {
+			fmt.Fprintf(o.ErrOut, "warning: Template parameter %q already defined, ignoring value from file %q", key, file)
+		}
+		return nil
+	})
+	if len(duplicatedKeys) != 0 {
+		return o.usageErrorFn(fmt.Sprintf("The following parameters were provided more than once: %s", strings.Join(duplicatedKeys.List(), ", ")))
 	}
 
 	var (
-		objects []runtime.Object
-		infos   []*resource.Info
+		err   error
+		infos []*resource.Info
 
 		client templateclient.TemplateInterface
 	)
 
-	if local {
+	if o.local {
 		// TODO: Change f.Object() so that it can fall back to local RESTMapper safely (currently glog.Fatals)
 	} else {
-		clientConfig, err := f.ToRESTConfig()
+		clientConfig, err := o.clientConfigFn()
```

Comment:
in some followup, we need to sweep commands to eliminate this construct.  We no longer hit the server to build these.